### DISCUSSION TITLE
Responsible for resolving any issues

### DISCUSSION THREAD

**Discussion by agjftucker** [Homebrew/discussions#1242] [Getting started]
It doesn't seem very fair to put this on me. I did not press RETURN when I got this warning. I pressed ESC, pretty much the opposite of RETURN. Yet the bloody thing proceeded regardless.

You are using macOS 10.13.
==> We (and Apple) do not provide support for this old version.
This installation may not succeed.
After installation, you will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or IRC. You are responsible for resolving any issues you experience
while you are running this old version.

Press RETURN to continue or any other key to abort

**Comment by SMillerDev** [comment]
You can just uninstall homebrew again https://docs.brew.sh/FAQ#how-do-i-uninstall-homebrew

If it didn't cancel as expected you can create an issue for that at https://github.com/Homebrew/install/issues

**Reply by agjftucker** [reply]
Thank you but I had a previously working version of Homebrew that I relied upon, so removing the whole thing will be a last resort. I will certainly make an issue for this, thank you for pointing me the way.

**Reply by SMillerDev** [reply]
Why did you rerun the installer if you had a working install? It doesn't do anything if you already have homebrew installed.

**Reply by agjftucker** [reply]
It was invoked by a script from a third party, this one in fact: https://raw.githubusercontent.com/leanprover-community/mathlib-tools/master/scripts/install_macos.sh

**Reply by agjftucker** [reply]
And it clearly did do quite a lot, removed a thousand of these things, installed a thousand of those things...

**Reply by SMillerDev** [reply]
In that case you can hold off on making an issue. It'll bypass any interaction inside a script so this is expected behavior.

**Reply by agjftucker** [reply]
Well it's not great behaviour. But you're saying there's nothing that can be done about it? My setup was rather delicate and I'm very worried it has been trampled over

**Reply by carlocab** [reply]
The correct place to file an issue is at https://github.com/leanprover-community/mathlib.

Their script should be testing whether `brew` is installed first before attempting to install it. Even better, you can open a PR; fixing it should be fairly easy.

**Reply by agjftucker** [reply]
> It'll bypass any interaction inside a script so this is expected behavior.

The script subsequently called other programs that I was able to cancel through keyboard interaction, so it must be possible?

> fixing it should be fairly easy.

I think you overestimate the technical skills of the average user! Certainly mine.

**Reply by carlocab** [reply]
> I think you overestimate the technical skills of the average user! Certainly mine.

If you look at the bottom of the script you'll see that it tests whether VS Code is installed before trying to use `brew` to install it. You can do pretty much the same thing at the top of the script.

**Reply by tlk** [reply]
> Their script should be testing whether `brew` is installed

Just out of curiosity, is it documented anywhere that the Homebrew installer script will wipe out an existing Homebrew installation?

**Reply by carlocab** [reply]
Does it wipe out an existing Homebrew installation?

**Reply by tlk** [reply]
I don't know - but it did something that the OP thought was undesirable. Is this documented somewhere?

**Reply by carlocab** [reply]
Sorry, but I still don't know what you're looking to be documented here. `Running the install script when you have an existing Homebrew install might make you feel bad.`?

**Reply by agjftucker** [reply]
> Does it wipe out an existing Homebrew installation?

Certainly it listed many hundreds of "Deleted Formulae" and "Deleted Casks" before I stopped it. I gave up on today after that so it remains to be seen what is the final impact on my working environment. Naturally I remain concerned given the warning "You will encounter build failures ... You are responsible for resolving any issues".

To be clear I think the warning would have been a perfectly adequate guard ... if it had responded to keyboard input as it promised. I'm still not clear why other programs called by this script were able to respond to keyboard input correctly, but for this one not doing so is "expected behaviour".
